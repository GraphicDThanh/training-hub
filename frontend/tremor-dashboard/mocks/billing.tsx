// Constants
import { AGGREGATION_TYPE } from "@/constants";
import { BillingInfoData } from "@/types";

export const MOCK_SALARY_DATA = [
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
];

export const MOCK_BILLING_CARD = {
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};

export const MOCK_BILLING_INFO: BillingInfoData = {
  id: "1",
  ownerName: "Oliver Liam",
  companyName: "Viking Burrito",
  cardLast4Digit: "1234",
  email: "oliver@burrito.com",
  vat: "FRB1235476",
};

export const MOCK_BILLING: BillingInfoData[] = [
  {
    id: "1",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "2",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "3",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
];
