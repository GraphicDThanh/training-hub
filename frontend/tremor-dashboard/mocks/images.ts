export const MOCK_URL_AVATAR =
  "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/product-details-1.f79d1e89.jpg";

export const MOCK_IMAGE_UPLOAD =
  "https://i.ibb.co/2WyPpQQ/0daa97f4b0dbe9a9686984982977da86-1.png";
