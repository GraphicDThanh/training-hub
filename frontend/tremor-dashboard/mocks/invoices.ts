import { Invoice } from "@/types";

export const MOCK_INVOICE_DATA = {
  id: 230220,
  createdAt: "06/03/2019",
  dueAt: "11/03/2019",
  bankInfo: {
    address: "St. Independence Embankment, 050105",
    city: "Bucharest",
    state: "Romania",
    phone: "+4 (074) 1090873",
  },
  customer: {
    fullName: "John Doe",
    address: "4006 Locust View Drive",
    city: "San Francisco",
    stateCode: "CA",
    state: "California",
  },
  products: [
    {
      id: 1,
      productName: "Premium Support",
      quantity: 1,
      price: 9,
    },
    {
      id: 2,
      productName: "Metro Dashboard",
      quantity: 3,
      price: 100,
    },
    {
      id: 3,
      productName: "Parts for service",
      quantity: 2,
      price: 10,
    },
  ],
  totalCost: 345,
};

export const MOCK_INVOICES: Invoice[] = [
  {
    id: 804322,
    createdAt: "March, 21, 2021",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 8908543,
    createdAt: "Apr, 22, 2022",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 415645,
    createdAt: "May, 23, 2023",
    invoicePrefix: "MS",
    totalCost: 180,
  },
  {
    id: 103578,
    createdAt: "Jun, 24, 2024",
    invoicePrefix: "QW",
    totalCost: 180,
  },
  {
    id: 126749,
    createdAt: "Jul, 25, 2025",
    invoicePrefix: "RV",
    totalCost: 180,
  },
];

export const MOCK_INVOICE_HEADER = {
  addressBank: "St. Independence Embankment, 050105",
  cityBank: "Bucharest",
  stateBank: "Romania",
  phoneBank: "+4 (074) 1090873",
  fullName: "John Doe",
  addressCustomer: "4006 Locust View Drive",
  cityCustomer: "San Francisco",
  stateCode: "CA",
  stateCustomer: "California",
};

//addressBank: string;
//cityBank: string;
//stateBank: string;
//phoneBank: string;
//fullName: string;
//addressCustomer: string;
//cityCustomer: string;
//stateCode: string;
//stateCustomer: string;

//id: number;
//createdAt: string;
//orderCode: string; }> & {
// id?: number | undefined;
//createdAt?: string | undefined;
//orderCode?: string | undefined;
