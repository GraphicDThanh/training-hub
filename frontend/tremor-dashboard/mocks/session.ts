import { User } from "@/types";

import * as SessionContext from "@/context/session";

export const mockUseSession = (user: User | null = null) => {
  jest.spyOn(SessionContext, "useSession").mockImplementation(() => ({
    user,
    setUser: () => {},
  }));
};
