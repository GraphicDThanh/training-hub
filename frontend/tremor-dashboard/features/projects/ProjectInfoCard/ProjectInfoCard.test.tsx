// ProjectInfoCard.test.tsx
import { render } from "@testing-library/react";

// Components
import ProjectInfoCard from "./ProjectInfoCard";

// Mocks
import { PROJECT_DATA } from "@/mocks";

const ProjectInfoCardProps = {
  projects: PROJECT_DATA,
};

const ProjectInfoCardComponent = () =>
  render(<ProjectInfoCard {...ProjectInfoCardProps} />);

describe("ProjectInfoCard Component", () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn();
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    });
    window.IntersectionObserver = mockIntersectionObserver;
  });

  it("render ProjectInfoCard component with snapshot correctly", () => {
    const { container } = ProjectInfoCardComponent();

    expect(container).toMatchSnapshot();
  });

  it("renders ProjectInfoCard component with mock data", () => {
    const { getByText } = ProjectInfoCardComponent();

    const projectNames = PROJECT_DATA.map(({ name }) => getByText(name));
    expect(projectNames.length).toBe(PROJECT_DATA.length);
  });
});
