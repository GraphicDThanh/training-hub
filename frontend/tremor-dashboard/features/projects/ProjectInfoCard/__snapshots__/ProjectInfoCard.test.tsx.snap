// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfoCard Component render ProjectInfoCard component with snapshot correctly 1`] = `
<div>
  <div
    class="tremor-Flex-root w-full flex-row justify-start items-start flex-wrap grid gap-x-6 gap-y-1 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3"
  >
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Slack Bot-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=128&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            1
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Slack Bot
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              If everything I did failed - which it doesn't, I think that it actually succeeds.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Premium Support-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="http://localhost/images/image-regular.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            2
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Premium Support
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              Pink is obviously a better color. Everyone’s born confident, and everything’s taken away from you.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fteam-1.cd822dc7.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Design Tools-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-asana.587ac859.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            3
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Design Tools
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              Constantly growing. We’re constantly making mistakes from which we learn and improve.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Looking Great-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-asana.587ac859.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            4
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Looking Great
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              You have the opportunity to play this game of life you need to appreciate every moment.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Developer First-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-asana.587ac859.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            5
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Developer First
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              For standing out. But the time is now to be okay to be the greatest you.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-start"
    >
      <div
        class="tremor-Card-root w-full text-left rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card dark:bg-dark-tremor-primary mx-auto py-1 pt-[5px] px-4 2xl:px-6 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border border-romance dark:border-grey relative mt-[40px] rounded-xl shadow"
      >
        <div>
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Product Development-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-asana.587ac859.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-start pb-3 pt-1"
        >
          <p
            class="text-tremor-default text-tremor-content capitalize font-normal dark:text-dark-tremor-content-romance pl-[5.5rem]"
          >
            Project #
            6
          </p>
          <h4
            class="w-full font-semibold text-md capitalize text-primary dark:text-dark-primary pl-[5.5rem] leading-6 line-clamp-2 items-center"
          >
            <a
              href="/projects"
            >
              Product Development
            </a>
          </h4>
          <div>
            <div
              class="text-sm text-tertiary my-4 font-light dark:text-dark-tremor-content-romance h-[2.8rem] line-clamp-2 [&>*>*]:!text-primary [&>*>*]:dark:!text-white"
            >
              We strive to embrace and drive change in our industry. We are happy to work at such a project.
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <a
              href="/projects"
            >
              <button
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand dark:text-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-faint box-shadow-transparent uppercase font-sans bg-inherit hover:bg-transparent hover:opacity-75 py-[7px] px-4"
              >
                <span
                  class="tremor-Button-text text-tremor-default whitespace-nowrap"
                >
                  <p
                    class="text-tremor-default font-bold text-seldom text-xs leading-[17px] dark:text-dark-tremor-content-seldom"
                  >
                    View Project
                  </p>
                </span>
              </button>
            </a>
            <div>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-end items-start mt-1 pl-1 h-[1.5rem] truncate"
              >
                <div
                  after-dynamic-value="Avatar 34"
                  class="inline-flex shadow-box-avatar rounded-full justify-center items-center overflow-hidden border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                >
                  <img
                    alt="Avatar 34"
                    class="object-cover border-white first:ml-0 border-2 border-solid ml-[-10px] w-[24px] h-[24px] min-w-[24px]"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="24"
                    src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fmarie.460e7723.jpg&w=48&q=75 2x"
                    style="color: transparent;"
                    width="24"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
